03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,21 +1,43 @@
 const Task = require('../models/Task')
+const asyncWrapper = require('../middleware/async')
+const {createCustomError} = require('../errors/custom-error')
 
-const getAllTasks = (req, res) => {
-    res.send('get all tasks')
-}
-const createTask = async (req, res) => {
-    const task = await Task.create(req.body)
-    res.status(201).json({task})
-}
-const getTask = (req, res) => {
-    res.json({id: req.params.id})
-}
-const updateTask = (req, res) => {
-    res.send('update task')
-}
-const deleteTask = (req, res) => {
-    res.send('delete task')
-}
+
+const getAllTasks = asyncWrapper (async (req, res) => {
+        const tasks = await Task.find({})
+        res.status(200).json({tasks})
+})
+const createTask = asyncWrapper (async (req, res) => {
+        const task = await Task.create(req.body)
+        res.status(201).json({task}) 
+})
```

Comment:
**Suggestion:** One improvement you could make here is to check that a name was provided. If not, you can return a custom error with status 400 (bad request) and a descriptive message.